systest/ak_test.go

```
@@ -99,27 +101,65 @@ func runTestSteps(t *testing.T, steps []string, akPath, akAddr string) {
 	var (
 		actionIndex int
 		ak          *akResult
-		httpResp    *string
+		pendingReq  *httpRequest
+		httpResp    *httpResponse
 	)
 	for i, step := range steps {
 		// Skip empty lines and comments.
 		if step == "" {
 			continue
 		}
 
-		// Actions: ak, http.
+		// Actions: ak, http, wait.
 		if actions.MatchString(step) {
+			// Before starting a new action, if there's a pending HTTP
+			// request, send it first. We implement it this way to
+			// support optional customizations below the action.
+			if pendingReq != nil {
+				resp, err := sendRequest(akAddr, *pendingReq)
+				if err != nil {
+					t.Errorf("line %d: %s", actionIndex+1, steps[actionIndex])
+					// Fail-fast, don't run subsequent test steps.
+					t.Fatalf("error: %v", err)
+				}
+				pendingReq = nil
+				httpResp = resp
+			}
+
+			// Now start with the new action, and store its result.
 			actionIndex = i
 			result, err := runAction(t, akPath, akAddr, step)
 			if err != nil {
 				t.Errorf("line %d: %s", i+1, step)
 				// Fail-fast, don't run subsequent test steps.
 				t.Fatalf("error: %v", err)
 			}
-			ak = result.(*akResult)
+
+			switch v := result.(type) {
+			case *akResult:
+				ak = v
+			case *httpRequest:
+				pendingReq = v
+			case string:
+				t.Log(v)
```

add default to not silently ignore unexpected?

---

done